sui_types/src/serialize.rs

```diff
@@ -85,65 +118,65 @@ pub fn serialize_transfer_transaction_into<W>(
 where
     W: std::io::Write,
 {
-    serialize_into(writer, &ShallowSerializedMessage::Transaction(value))
+    serialize_into(writer, None, &ShallowSerializedMessage::Transaction(value))
```

Comment:
Maybe it would be worth having a serialize_with_header, instead of changing everything to have None, to support the uncommon (I hope) case of headers.